src/libraries/System.Security.Cryptography.Cose/src/System/Security/Cryptography/Cose/CoseMessage.cs

```diff
@@ -176,17 +251,63 @@ private static void DecodeBucket(CborReader reader, CoseHeaderMap headerParamete
                 return reader.ReadByteString();
             }
 
-            throw new CryptographicException(SR.Format(SR.DecodeSign1ErrorWhileDecoding, SR.DecodeSign1PayloadWasIncorrect));
+            throw new CryptographicException(SR.Format(SR.DecodeErrorWhileDecoding, SR.DecodeSign1PayloadWasIncorrect));
         }
 
         private static byte[] DecodeSignature(CborReader reader)
         {
             return reader.ReadByteString();
         }
 
-        internal static void AppendToBeSigned(Span<byte> buffer, IncrementalHash hasher, string context, ReadOnlySpan<byte> encodedProtectedHeader, ReadOnlySpan<byte> contentBytes, Stream? contentStream, HashAlgorithmName hashAlgorithm)
+        private static void DecodeCoseSignaturesArray(CborReader reader, out List<CoseSignature> signatures, byte[] bodyProtected, byte[]? content)
         {
-            int bytesWritten = CreateToBeSigned(buffer, context, encodedProtectedHeader, ReadOnlySpan<byte>.Empty);
+            int? signaturesLength = reader.ReadStartArray();
+
+            if (signaturesLength.GetValueOrDefault() < 1)
+            {
+                throw new CryptographicException(SR.Format(SR.DecodeErrorWhileDecoding, SR.MultiSignMessageMustCarryAtLeastOneSignature));
+            }
+
+            signatures = new List<CoseSignature>(signaturesLength!.Value);
+
+            for (int i = 0; i < signaturesLength; i++)
+            {
+                int? length = reader.ReadStartArray();
+
+                if (length != 3)
```

Comment:
We have a named const for this, we should use it.

```suggestion
                if (length != CoseMultiSignMessage.CoseSignatureArrayLength)
```